PopPUNK/__main__.py

```diff
@@ -114,11 +114,34 @@ def get_options():
 
     # model fitting
     modelGroup = parser.add_argument_group('Model fit options')
-    modelGroup.add_argument('--K', help='Maximum number of mixture components [default = 2]', type=int, default=2)
-    modelGroup.add_argument('--D', help='Maximum number of clusters in DBSCAN fitting [default = 100]', type=int, default=100)
-    modelGroup.add_argument('--min-cluster-prop', help='Minimum proportion of points in a cluster '
-                                                        'in DBSCAN fitting [default = 0.0001]', type=float, default=0.0001)
-    modelGroup.add_argument('--threshold', help='Cutoff if using --fit-model threshold', type=float)
+    modelGroup.add_argument('--model-subsample',
+                            help='Number of pairwise distances used to fit model [default = 100000]',
+                            type=int,
+                            default=100000)
+    modelGroup.add_argument('--assign-subsample',
+                            help='Number of pairwise distances in each assignment batch [default = 5000]',
+                            type=int,
+                            default=5000)
```

Comment:
Will these ever likely be changed? I think can set as constants if not (seems like these defaults should always work?)

Reply:
There are times when it is helpful to change them - the `--assign-subsample` in particular controls the flow of data into GPU memory, which can either be horrendously slow, or overload the device memory (when combined with the size of the model in GPU memory). The subsampling is helpful I think, because sometimes it is difficult to get enough representation of the between strain distribution when one strain is heavily sampled.

Reply:
Fair enough! Perhaps worth putting this description in the GPU page of the docs?

Reply:
Added in [6795025](https://github.com/bacpop/PopPUNK/pull/302/commits/6795025be0c25c72ff99a5731954f5c28169f7b5), and tests added in [9a229c6](https://github.com/bacpop/PopPUNK/pull/302/commits/9a229c67bf31b78ae7ecc249de9ba5892ed355e9) - anything else missing, do you think?

Reply:
Looks good!